stdlib/function.ncl

```diff
@@ -7,5 +7,35 @@
     const : forall a b. a -> b -> a
     | doc "The const function, results in the first argument."
     = fun x y => x,
+
+    compose : forall a b c. (b -> c) -> (a -> b) -> (a -> c)
+    | doc "Function composition, right to left."
+    = fun g f x => x |> f |> g,
+
+    flip : forall a b c. (a -> b -> c) -> b -> a -> c
+    | doc "Flip the argument order for a curried function."
+    = fun f x y => f y x,
+
+    fst : forall a b. a -> b -> a
+    | doc "Always returns the first argument."
+    = const,
+
+    snd : forall a b. a -> b -> b
+    | doc "Always returns the second argument."
+    = flip const,
+
+    pipe | Dyn -> Array (Dyn -> Dyn) -> Dyn
```

Comment:
I wonder if `apply_all` wouldn't be more explicit, as a name. But this is just a remark, we can bikeshed later.

Reply:
Also, is there a reason to use a contract instead of a type annotation here?

Reply:
I seem to remember that we were concerned about changing types through different stages of the pipeline. But that doesn't seem to actually be a problem, `pipe : forall a. a -> Array (a -> a) -> a` works just fine for the examples.